Autopush broken due to conversion from master to main on some upstream repos

```
Cloning into bare repository 'srcs/pythondata-cpu-serv'...
fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
Traceback (most recent call last):
  File "./update.py", line 559, in <module>
    sys.exit(main(sys.argv[0], sys.argv[1:]))
  File "./update.py", line 511, in main
    get_src(m)
  File "./update.py", line 193, in get_src
    git_hash = get_hash(module_data['branch'], env)
  File "./update.py", line 125, in get_hash
    env=env).decode('utf-8').strip()
  File "/usr/lib/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['git', 'rev-parse', 'master']' returned non-zero exit status 128.
Error: Process completed with exit code 1.
```

--- a/modules.ini
+++ b/modules.ini
@@ -1,5 +1,6 @@
 [DEFAULT]
 branch = master
+submodule = False
 
 # Verilog for various CPU cores
 # -------------------------------------

--- a/modules.ini
+++ b/modules.ini
@@ -127,9 +127,10 @@ license_spdx = Apache-2.0
 type = software
 human_name = picolibc - a C library designed for embedded 32- and 64- bit systems.
 src = https://github.com/picolibc/picolibc
+branch = main
 contents = data
 license = License :: OSI Approved :: BSD License
-license_spdx = BSD3-1
+license_spdx = BSD-3-Clause
 
 [tapcfg]
 type = misc

--- a/modules.ini
+++ b/modules.ini
@@ -130,6 +130,7 @@ human_name = picolibc - a C library designed for embedded 32- and 64- bit system
 src = https://github.com/picolibc/picolibc
 branch = main
 contents = data
+submodule = True
 license = License :: OSI Approved :: BSD License
 license_spdx = BSD-3-Clause
 

--- a/modules.ini
+++ b/modules.ini
@@ -82,6 +82,7 @@ git_hash = 5f0c7a7faf6b65c907a93be6e3723e297d37ee71
 type = cpu
 human_name = serv
 src = https://github.com/olofk/serv
+branch = main
 contents = verilog
 license = License :: OSI Approved :: ISC License (ISCL)
 license_spdx = ISC

--- a/update.py
+++ b/update.py
@@ -73,8 +73,13 @@ def github_repo(g, module_data):
 def download(module_data):
     out_path = os.path.join('repos',module_data['repo'])
     if not os.path.exists(out_path):
+
+        clone_opts = ''
+        if module_data.getboolean('submodule'):
+            clone_opts = '--recursive'
+
         subprocess_check_call(
-            ["git", "clone", module_data['repo_url'], out_path])
+            ["git", "clone", clone_opts, module_data['repo_url'], out_path])
     else:
         dotgit = os.path.join(out_path, '.git')
         assert os.path.exists(dotgit), dotgit

--- a/update.py
+++ b/update.py
@@ -377,17 +377,26 @@ Updated using {tool_version} from https://github.com/litex-hub/litex-data-auto
 
     # Run the git subtree command
     if 'src' in module_data:
-        if os.path.exists(os.path.join(repo_dir, module_data['dir'])):
-            subtree_cmd = 'pull'
+        if module_data.getboolean('submodule'):
+            if os.path.exists(os.path.join(repo_dir, module_data['dir'])):
+                cmd = 'git submodule update --remote --merge'
+            else:
+                submodule_cmd = 'git submodule add {} {}'.format(
+                        module_data['src'], module_data['dir'])
+            print(cmd)
+            subprocess_check_call(cmd.split(), cwd=repo_dir)
         else:
-            subtree_cmd = 'add'
-        cmd = [
-            'git', 'subtree', subtree_cmd,
-            '-P', module_data['dir'],
-            module_data['src_local'], module_data['data_git_hash'],
-        ]
-        print(cmd)
-        subprocess_check_call(cmd, cwd=repo_dir)
+            if os.path.exists(os.path.join(repo_dir, module_data['dir'])):
+                subtree_cmd = 'pull'
+            else:
+                subtree_cmd = 'add'
+            cmd = [
+                'git', 'subtree', subtree_cmd,
+                '-P', module_data['dir'],
+                module_data['src_local'], module_data['data_git_hash'],
+            ]
+            print(cmd)
+            subprocess_check_call(cmd, cwd=repo_dir)
 
 
 def push(module_data):

--- a/update.py
+++ b/update.py
@@ -390,6 +390,18 @@ Updated using {tool_version} from https://github.com/litex-hub/litex-data-auto
                         module_data['src'], module_data['dir'])
             print(cmd)
             subprocess_check_call(cmd.split(), cwd=repo_dir)
+            # submodule bump does not commit by itself
+            subprocess_check_call(['git', 'add', '.'], cwd=repo_dir)
+            with tempfile.NamedTemporaryFile() as f:
+                f.write("""\
+Bump {dir} submodule to {data_git_hash}
+
+Updated using {tool_version} from https://github.com/litex-hub/litex-data-auto
+""".format(**module_data).encode('utf-8'))
+                f.flush()
+                f.name
+                subprocess_check_call(['git', 'commit', '-F', f.name], cwd=repo_dir)
+
         else:
             if os.path.exists(os.path.join(repo_dir, module_data['dir'])):
                 subtree_cmd = 'pull'

